src/auth/interface.js

```diff
@@ -117,78 +120,125 @@ export function resetAuthService() {
+ * ```
+ *
+ * @param {string} redirectUrl The URL the user should be redirected to after logging in.
  */
 export function getLoginRedirectUrl(redirectUrl) {
   return service.getLoginRedirectUrl(redirectUrl);
 }
 
 /**
+ * Redirects the user to the login page.
  *
+ * @param {string} redirectUrl The URL the user should be redirected to after logging in.
  */
 export function redirectToLogin(redirectUrl) {
   return service.redirectToLogin(redirectUrl);
 }
 
 /**
+ * Builds a URL to the logout page with a post-logout redirect URL attached as a query parameter.
+ *
+ * ```
+ * const url = getLogoutRedirectUrl('http://localhost/mypage');
+ * console.log(url); // http://localhost/logout?next=http%3A%2F%2Flocalhost%2Fmypage
+ * ```
  *
+ * @param {string} redirectUrl The URL the user should be redirected to after logging out.
  */
 export function getLogoutRedirectUrl(redirectUrl) {
   return service.getLogoutRedirectUrl(redirectUrl);
 }
 
 /**
+ * Redirects the user to the logout page.
  *
+ * @param {string} redirectUrl The URL the user should be redirected to after logging out.
  */
 export function redirectToLogout(redirectUrl) {
   return service.redirectToLogout(redirectUrl);
 }
 
 /**
+ * If it exists, returns the user data representing the currently authenticated user. If the
+ * user is anonymous, returns null.
  *
+ * @returns {UserData|null}
  */
 export function getAuthenticatedUser() {
   return service.getAuthenticatedUser();
 }
 
 /**
+ * Sets the authenticated user to the provided value.
  *
+ * @param {UserData} authUser
+ * @emits AUTHENTICATED_USER_CHANGED
  */
 export function setAuthenticatedUser(authUser) {
   service.setAuthenticatedUser(authUser);
   publish(AUTHENTICATED_USER_CHANGED);
 }
 
 /**
+ * Reads the authenticated user's access token. Resolves to null if the user is
+ * unauthenticated.
  *
+ * @returns {Promise<UserData>|Promise<null>} Resolves to the user's access token if they are
+ * logged in.
  */
 export async function fetchAuthenticatedUser() {
   return service.fetchAuthenticatedUser();
 }
 
 /**
+ * Ensures a user is authenticated. It will redirect to login when not
+ * authenticated.
  *
+ * @param {string} [redirectUrl=config.BASE_URL] to return user after login when not
+ * authenticated.
+ * @returns {Promise<UserData>}
  */
 export async function ensureAuthenticatedUser(redirectUrl) {
   return service.ensureAuthenticatedUser(redirectUrl);
 }
 
 /**
+ * Fetches additional user account information for the authenticated user and merges it into the
+ * existing authenticatedUser object, available via getAuthenticatedUser().
+ *
+ * ```
+ *  console.log(authenticatedUser); // Will be sparse and only contain basic information.
+ *  await hydrateAuthenticatedUser()
+ *  const authenticatedUser = getAuthenticatedUser();
+ *  console.log(authenticatedUser); // Will contain additional user information
+ * ```
  *
+ * @returns {Promise<null>}
```

Comment:
not sure if hydrate is expected to fail and if a boolean response instead of null would be better? if that has not come up we can ignore this note :)

Reply:
`hydrateAuthenticatedUser` doesn't actually resolve its promise with any data; any errors will get thrown/handled as part of the promise itself.  So `null` is pretty much correct, I think. :)